joern-cli/frontends/pysrc2cpg/src/test/scala/io/joern/pysrc2cpg/config/VenvExcludeTests.scala

```diff
@@ -0,0 +1,124 @@
+package io.joern.pysrc2cpg.config
+
+import better.files.File
+import io.joern.pysrc2cpg.Py2CpgOnFileSystem
+import io.joern.pysrc2cpg.Py2CpgOnFileSystemConfig
+import io.joern.x2cpg.ValidationMode
+import io.joern.x2cpg.X2Cpg.newEmptyCpg
+import io.shiftleft.semanticcpg.language.*
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.prop.TableDrivenPropertyChecks
+import org.scalatest.wordspec.AnyWordSpec
+import org.scalatest.BeforeAndAfterAll
+
+import java.nio.file.Path
+import java.nio.file.Paths
+import java.util.regex.Pattern
+
+class VenvExcludeTests extends AnyWordSpec with Matchers with TableDrivenPropertyChecks with BeforeAndAfterAll {
+
+  private val testFiles: List[String] = List(
+    ".venv/a1.py",
+    ".venv/sub/s1.py",
+    ".venv/pyvenv.cfg",
+    ".venv2/a2.py",
+    ".venv2/sub/s2.py",
+    ".venv2/pyvenv.cfg",
+    "folder/b.py",
+    "folder/c.py",
+    "main.py"
+  )
+
+  private val projectUnderTest: File = {
+    val dir = File.newTemporaryDirectory("pysrc2cpgTestsExcludeTest")
+    testFiles.foreach { testFile =>
+      val file = dir / testFile
+      file.createIfNotExists(createParents = true)
+    }
+    dir
+  }
+
+  override def afterAll(): Unit = projectUnderTest.delete(swallowIOExceptions = true)
+
+  private def testWithArguments(ignoreVenvDir: Boolean, venvDirs: Seq[String], expectedFiles: Set[String]): Unit = {
+    File.usingTemporaryDirectory("pysrc2cpgTests") { tmpDir =>
+      val config = Py2CpgOnFileSystemConfig()
+        .withInputPath(projectUnderTest.toString)
+        .withOutputPath(tmpDir.toString)
+        .withIgnoreVenvDir(ignoreVenvDir)
+        .withVenvDirs(venvDirs.map(Paths.get(_)))
+      val cpg = new Py2CpgOnFileSystem().createCpg(config).getOrElse(fail("No CPG was generated!"))
+      cpg.file.name.l should contain theSameElementsAs expectedFiles.map(_.replace("/", java.io.File.separator))
+    }
+  }
+
+  "Using different excludes via program arguments" should {
+
+    val testInput = Table(
+      // -- Header for naming all test parameters
+      ("statement", "ignoreVenvDir", "venvDirs", "expectedResult"),
+      // --
+      // Test for --ignoreVenvDir == false:
+      (
+        "exclude nothing if ignoreVenvDir is false",
+        false,
+        Seq.empty,
+        Set(
+          ".venv/a1.py",
+          ".venv/sub/s1.py",
+          ".venv2/a2.py",
+          ".venv2/sub/s2.py",
+          "folder/b.py",
+          "folder/c.py",
+          "main.py"
+        )
+      ),
```

Comment:
doesn't really matter, but we could write these tests as

```scala
"exclude nothing if ignoreVenvDir is false" should {
   assert(testWithArguments(false, Seq.empty, Seq(...)))
  }
}
```

should be a little shorter in the end, and result in somewhat nicer errors when the tests fail

Reply:
OK, leave it as-is then.

I've had some problems figuring out which table entry was failing for some scalatest tests in the past, so my knee-jerk reaction is to avoid them. But this might very well not apply here.